packages/plugin-core/src/commands/MergeNoteCommand.ts

```diff
@@ -0,0 +1,453 @@
+          const data: NoteLookupProviderSuccessResp = event.data;
+          await this.prepareProxyMetricPayload({
+            sourceNote: activeNote,
+            destNote: data.selectedItems[0],
+          });
+          resolve({
+            sourceNote: activeNote,
+            destNote: data.selectedItems[0],
+          });
+        },
+      });
+      const showOpts: CreateQuickPickOpts & {
+        nonInteractive?: boolean | undefined;
+        initialValue?: string | undefined;
+        provider: ILookupProviderV3;
+      } = {
+        title: "Select merge destination note",
+        placeholder: "note",
+        provider,
+      };
+      if (opts?.dest) {
+        showOpts.initialValue = opts.dest;
+      }
+      if (opts?.noConfirm) {
+        showOpts.nonInteractive = opts.noConfirm;
+      }
+      controller.show(showOpts);
+    });
+  }
+
+  private async prepareProxyMetricPayload(opts: {
+    sourceNote: NoteProps | undefined;
+    destNote: NoteProps | undefined;
+  }) {
+    const ctx = `${this.key}:prepareProxyMetricPayload`;
+    const { sourceNote, destNote } = opts;
+    if (sourceNote === undefined || destNote === undefined) {
+      // source or dest note undefined, this could be from cancellation.
+      // just return.
+      return;
+    }
+    const sourceBasicStats = StatisticsUtils.getBasicStatsFromNotes([
+      sourceNote,
+    ]);
+    const destBasicStats = StatisticsUtils.getBasicStatsFromNotes([destNote]);
+    if (sourceBasicStats === undefined || destBasicStats === undefined) {
+      this.L.error({ ctx, message: "failed to get basic stats from notes." });
+      return;
+    }
+
+    const { numChildren, numLinks, numChars, noteDepth } = sourceBasicStats;
+    const {
+      numChildren: destNumChildren,
+      numLinks: destNumLinks,
+      numChars: destNumChars,
+      noteDepth: destNoteDepth,
+    } = destBasicStats;
+
+    const sourceTraits = sourceNote.traits;
+    const destTraits = destNote.traits;
+
+    const engine = this.extension.getEngine();
+    this._proxyMetricPayload = {
+      command: this.key,
+      numVaults: engine.vaults.length,
+      numChildren,
+      numLinks,
+      numChars,
+      noteDepth,
+      traits: sourceTraits || [],
+      extra: {
+        destNumChildren,
+        destNumLinks,
+        destNumChars,
+        destNoteDepth,
+        destTraits: destTraits || [],
+      },
+    };
+  }
+
+  /**
+   * Given a source note and destination note,
+   * append the entire body of source note to the destination note.
+   * @param sourceNote Source note
+   * @param destNote Dest note
+   */
+  private async appendNote(opts: {
+    sourceNote: NoteProps;
+    destNote: NoteProps;
+  }) {
+    const { sourceNote, destNote } = opts;
+    // grab body from current active note
+    const appendPayload = sourceNote.body;
+
+    // append to end
+    const destBody = destNote.body;
+    const newBody = `${destBody}\n\n${appendPayload}`;
```

Comment:
should we have a line here saying the note was merged? if we go with evernote behavior, it would be

```
{body of previous note}
\n
---
# {title of merged note}
\n
{body of new note)
``

Reply:
that's a good idea.